com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/VariablesRequestHandler.java

```diff
@@ -236,15 +246,36 @@
                 }
             }
 
+            String evaluateName = null;
+            if (javaVariable.evaluateName == null || (containerEvaluateName == null && containerNode.getProxiedVariable() instanceof ObjectReference)) {
+                // Disable evaluate on the method return value.
+                evaluateName = null;
+            } else if (isUnboundedTypeContainer && !containerNode.isIndexedVariable()) {
+                // The type name returned by JDI is the binary name, which uses '$' as the separator of
+                // inner class e.g. Foo$Bar. But the evaluation expression only accepts using '.' as the class
+                // name separator.
+                String typeName = ((ObjectReference) containerNode.getProxiedVariable()).referenceType().name();
+                typeName = typeName.replaceAll("\\$", ".");
```

Comment:
Considering add some comments that this replacement will also possibly change the $ in class name?

Reply:
done.